VCFv4.3.tex

```diff
@@ -503,41 +503,100 @@ \subsection{VCF tag naming conventions}
-  \item INS: Insertion of novel sequence relative to the reference
-  \item DUP: Region of elevated copy number relative to the reference
-  \item INV: Inversion of reference sequence
-  \item CNV: Copy number variable region (may be both deletion and duplication)
+\item [\null] INS:  Insertion relative to the reference
+	\begin{itemize}
+	\item [\null] MEI:  Mobile element insertion relative to the reference
+		\begin{itemize}
+		\item [\null] ALU:  Alu mobile element insertion relative to the reference
+		\item [\null] LINE1:  LINE1 mobile element insertion relative to the reference
+		\item [\null] SVA:  SVA mobile element insertion relative to the reference
+		\item [\null] HERV:  HERV mobile element insertion relative to the reference
+		\end{itemize}
+	\item [\null] RET:  Retrotransposon insertion relative to the reference
+	\end{itemize}
+\item [\null] DEL:  Deletion relative to the reference
+	\begin{itemize}
+	\item [\null] -MEI:  Absence of a mobile element present in the reference
+		\begin{itemize}
+		\item [\null] -ALU:  Absence of a Alu mobile element present in the reference
+		\item [\null] -LINE1:  Absence of a LINE1 mobile element present in the reference
+		\item [\null] -SVA:  Absence of a SVA mobile element present in the reference
+		\item [\null] -HERV:  Absence of a HERV mobile element present in the reference
+		\end{itemize}
+	\item [\null] -RET:  Absence of a retrotransposon present in the reference
+	\end{itemize}
+\item [\null] TRA:  Translocation
+\item [\null] STR:  Short tandem repeat variation
+\item [\null] INV:  Inversion relative to the reference
+\item [\null] DUP:  Duplication
+	\begin{itemize}
+	\item [\null] TANDUP:  Tandem duplication (one of Direct, Inverted-upstream, or Inverted-downstream)
+	\item [\null] DISDUP:  Dispersed duplication
+	\end{itemize}
+\item [\null] DELINS:  Deletion-insertion (replacement of reference sequence with sequence of different content or length)
+\item [\null] OTHER:  Complex variant not covered elsewhere
+	\begin{itemize}
+	\item [\null] DEL\_DUP:  Deletion + Duplication
+	\item [\null] DEL\_INV:  Deletion + Inversion
+	\item [\null] DUP\_INV:  Duplication + Inversion
+	\item [\null] DEL\_DUP\_INV:  Deletion + Duplication + Inversion
+	\end{itemize}
+\item [\null] CNV:  Copy number variation\vspace{0.3cm}
 \end{itemize}
-
+TRA (Translocation):  A VCF translocation record describes the juxtaposition of two DNA segments from disparate genomic locations (from the same or from different chromosomes). It must include two genomic coordinates, each accompanied by an orientation (see STRAND below). A VCF translocation record only describes the immediate molecular context of the novel adjacency generated by the translocation event. It should not be confused with the traditional cytogenetic notion of translocation, which is understood to imply that the remainder of the chromosome arms beyond the specified breakpoints are also involved.\vspace{0.3cm}\\
```

Comment:
>it is in fairly wide usage for describing translocations in VCF (whereas BND is probably less so, at least in the VCF I have encountered)
>is supported by at least one major SV caller (DELLY)

DELLY changed from TRA to BND notation back in March this year so as to be VCF compliant (see https://github.com/dellytools/delly/issues/59 ). Do you know which other callers write TRA records? TIGRA uses CTX and CHR2 with a similar meaning to DELLY's TRA. As far as I can tell, LUMPY, GRIDSS, manta, and CREST are all compliant. Pindel deviates from the current specs with its use of RPL (replacement) for del/ins events.

>  in practice it is unwieldy and complicated to implement. If BND can describe all events, why have any other SVTYPEs at all?

I ask myself the same question. I believe it is because VCF was designed for SNVs and evolved over time to support single-loci events then finally arbitrarily complex structural variants through the use of BND notation. VCF is ill-suited to representing SVs but it's still a whole lot better than every tool writing their own custom tsv/csv format.

If you're handling SVs in general, you can't get away from having some events in BND notation. I agree it is unwieldy and complicated, but if the analysis/software already has to deal with BND events, adding additional event types on top of this just makes processing SVs even more messy and complicated.

> @d-cameron , do you agree that TRA serves the same sort of purpose as all the other, non-BND SVTYPEs?

Adding it to the specifications would serve the same purpose as all the other SVTYPEs (except CNV, although even those can now be represented, albeit extremely unintuitively, using breakend notation): as a short-hand to an equivalent set of BND records. The key difference between the existing types and TRA is that all the existing types are all single-loci events.


As far as I can tell, the proposed TRA notation is just an alternate representation of information contained in a BND record. Why not just use a BND record? How is the specifications enhanced by requiring all implementing program to support both BND and TRA notations?

Note that BND records were intentionally defined to be paired and have a record at both locations so the events involving any given loci could be found by querying that loci. Yes this is messy, but programs already have to deal with this for BND events and it would be inconsistent if the proposed TRA records were not also paired.